src/modules/users/components/languages-list/languages-list.tsx

```diff
@@ -0,0 +1,117 @@
+    const { t } = useTranslation();
+
+    const { data, loading } = useUserLanguages({ userId });
+
+    const handleLanguageSelected = (lang: LanguageProficiency) => {
+        updateLanguage(lang);
+    };
+
+    const [openLanguageDialog] = useLanguagesDialog();
+
+    const [updateMutation, { loading: loadingUpdate }] = useUpdateUserLanguage();
+    const [createMutation, { loading: loadingAdd }] = useCreateUserLanguage();
+
+    const existingLanguageLabels = useMemo(
+        () => (data ? data?.profile.languages.map((item) => item.name) : []),
+        [data]
+    );
+
+    const addLanguage = useCallback(
+        () =>
+            openLanguageDialog({
+                title: t("languages.addLanguage"),
+                onConfirm: async ({ name, proficiency }: LanguagesFormSchemaType) => {
+                    await createMutation({
+                        variables: {
+                            language: {
+                                userId,
+                                name,
+                                proficiency: proficiency as Proficiency,
+                            },
+                        },
+                    });
+                },
+                userId,
+                existingLanguages: existingLanguageLabels,
+                initialValues: {
+                    name: "",
+                    proficiency: "",
+                },
+                isSubmitting: loadingAdd,
+            }),
+        [openLanguageDialog, userId, loadingAdd, existingLanguageLabels]
+    );
+
+    const updateLanguage = useCallback(
+        (selectedLanguage: LanguageProficiency) =>
+            openLanguageDialog({
+                title: t("languages.updateLanguage"),
+                onConfirm: async ({ name, proficiency }: LanguagesFormSchemaType) => {
+                    await updateMutation({
+                        variables: {
+                            language: {
+                                userId,
+                                name,
+                                proficiency: proficiency as Proficiency,
+                            },
+                        },
+                    });
+                },
+                userId,
+                selectedLanguage,
+                existingLanguages: existingLanguageLabels,
+                initialValues: {
+                    name: selectedLanguage.name,
+                    proficiency: selectedLanguage.proficiency,
+                },
+                isSubmitting: loadingUpdate,
+            }),
+        [openLanguageDialog, userId, loadingUpdate, existingLanguageLabels]
+    );
+
+    if (loading) return <Spin size="large" />;
+
+    return (
+        <Flex vertical className={styles.list}>
+            {canEdit ? (
+                <Button size={"large"} type="text" onClick={addLanguage}>
+                    <PlusOutlined />
+                    {t("languages.addLanguage")}
+                </Button>
+            ) : null}
```

Comment:
```suggestion
            {canEdit && (
                <Button size={"large"} type="text" onClick={addLanguage}>
                    <PlusOutlined />
                    {t("languages.addLanguage")}
                </Button>
            ) }
```

avoid using ternaries

Reply:
fixed